src/rust/src/lib.rs

```diff
@@ -195,12 +191,11 @@ fn decode_vectorized_(what: Either<Strings, List>, engine: Robj) -> Robj {
             .set_class(&["blob", "vctrs_list_of", "vctrs_vctr", "list"])
             .unwrap(),
         Either::Right(r) => {
-            if !r.inherits("blob") {
-                throw_r_error("Expected a character vector or an object of class `blob`")
-            }
             r.into_iter()
                 .map(|(_, b)| {
-                    if b.is_null() {
+                    if !b.inherits("raw") {
+                        Rstr::na().into_robj()
+                    } else if b.is_null() {
                         ().into_robj()
                     } else {
                         let raw: Raw = b.try_into().unwrap();
```

Comment:
See related https://github.com/extendr/extendr/issues/687.

Here we should use try_from() and match as it will catch when it is not a `Raw` and I suppose it is also more "idiomatic" rust as well.

```rust
   let raw = Raw::try_from(b);
                    match raw {
                        Ok(r) => {
                            let decoded = eng.decode(r.as_slice());
                            match decoded {
                                Ok(d) => Raw::from_bytes(&d).into_robj(),
                                Err(_) => ().into_robj(),
                            }
                        },
                        Err(_) => ().into_robj(),
                    }
                })
```